File: langs/java/backend/src/main/java/dev/argon/nobleidl/compiler/api/EsexprRecordOptions.java
package dev.argon.nobleidl.compiler.api;

import dev.argon.esexpr.Constructor;
import dev.argon.esexpr.ESExprCodec;
import dev.argon.esexpr.ESExprCodecGen;
import dev.argon.esexpr.Keyword;

@ESExprCodecGen
@Constructor("record-options")
public record EsexprRecordOptions(
	@Keyword
	String constructor
) {
	public static ESExprCodec<EsexprRecordOptions> codec() {
		return EsexprRecordOptions_CodecImpl.INSTANCE;
	}
}

File: langs/scala/compiler-api/src/main/scala/nobleidl/compiler/api/EsexprRecordOptions.scala
package nobleidl.compiler.api
@_root_.esexpr.constructor("record-options")
final case class EsexprRecordOptions(
  @_root_.esexpr.keyword("constructor")
  constructor: _root_.nobleidl.core.String,
) derives _root_.esexpr.ESExprCodec, _root_.scala.CanEqual
object EsexprRecordOptions {
  def javaAdapter(): _root_.nobleidl.core.JavaAdapter[_root_.nobleidl.compiler.api.EsexprRecordOptions, _root_.dev.argon.nobleidl.compiler.api.EsexprRecordOptions] =
    new _root_.nobleidl.core.JavaAdapter[_root_.nobleidl.compiler.api.EsexprRecordOptions, _root_.dev.argon.nobleidl.compiler.api.EsexprRecordOptions] {
      override def toJava(s_value: _root_.nobleidl.compiler.api.EsexprRecordOptions): _root_.dev.argon.nobleidl.compiler.api.EsexprRecordOptions = {
        new _root_.dev.argon.nobleidl.compiler.api.EsexprRecordOptions(
          _root_.nobleidl.core.String.javaAdapter().toJava(s_value.constructor),
        )
      }
      override def fromJava(j_value: _root_.dev.argon.nobleidl.compiler.api.EsexprRecordOptions): _root_.nobleidl.compiler.api.EsexprRecordOptions = {
        _root_.nobleidl.compiler.api.EsexprRecordOptions(
          _root_.nobleidl.core.String.javaAdapter().fromJava(j_value.constructor().nn),
        )
      }
    }
}
